File: app/src/main/java/com/qianlei/jiaowu/core/db/MyDataBase.java
package com.qianlei.jiaowu.core.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.qianlei.jiaowu.core.db.dao.ExamDao;
import com.qianlei.jiaowu.core.db.dao.ScoreDao;
import com.qianlei.jiaowu.core.db.dao.SubjectDao;
import com.qianlei.jiaowu.entity.Examination;
import com.qianlei.jiaowu.entity.Score;
import com.qianlei.jiaowu.entity.Subject;

/**
 * @author qianlei
 */
@Database(entities = {Examination.class, Score.class, Subject.class}, version = 1, exportSchema = false)
public abstract class MyDataBase extends RoomDatabase {
    private static MyDataBase INSTANCE;

    public static synchronized MyDataBase getDatabase(Context context) {
        if (INSTANCE == null) {
            synchronized (MyDataBase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            MyDataBase.class, "database").build();
                }
            }
        }
        return INSTANCE;
    }

    public abstract ExamDao getExamDao();

    public abstract ScoreDao getScoreDao();

    public abstract SubjectDao getSubjectDao();
}

File: app/src/main/java/com/qianlei/jiaowu/db/MyDataBase.kt
package com.qianlei.jiaowu.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.qianlei.jiaowu.db.dao.ExamDao
import com.qianlei.jiaowu.db.dao.ScoreDao
import com.qianlei.jiaowu.db.dao.SubjectDao
import com.qianlei.jiaowu.entity.Examination
import com.qianlei.jiaowu.entity.Score
import com.qianlei.jiaowu.entity.Subject

/**
 * @author qianlei
 */
@Database(entities = [Examination::class, Score::class, Subject::class], version = 1, exportSchema = false)
abstract class MyDataBase : RoomDatabase() {
    abstract fun examDao(): ExamDao
    abstract fun scoreDao(): ScoreDao
    abstract fun subjectDao(): SubjectDao

    companion object {
        private var INSTANCE: MyDataBase? = null
        fun getDatabase(context: Context?): MyDataBase {
            if (INSTANCE == null) {
                synchronized(MyDataBase::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(context!!.applicationContext,
                                MyDataBase::class.java, "database")
                                .build()
                    }
                }
            }
            return INSTANCE!!
        }
    }
}